### PR TITLE
feat(serviceRelease): get service type

### PR DESCRIPTION
**Pull request by qxz2mqe**
added new endpoint to get service type data
Refs: CPLP-2347, CPLP-2293

### CODE REVIEW

File: src/portalbackend/PortalBackend.DBAccess/Models/ServiceTypeData.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+using Org.Eclipse.TractusX.Portal.Backend.PortalBackend.PortalEntities.Enums;
+
+namespace Org.Eclipse.TractusX.Portal.Backend.PortalBackend.DBAccess.Models;
+
+public record ServiceTypeData(int ServiceTypeId, string Name);
```
Comment (Phil91) [inline src/portalbackend/PortalBackend.DBAccess/Models/ServiceTypeData.cs:24]: ServiceTypeId need to be of ServiceTypeId

File: src/marketplace/Services.Service/BusinessLogic/ServiceReleaseBusinessLogic.cs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            result.Id,
+            result.Title ?? Constants.ErrorString,
+            result.ServiceTypeIds,
+            result.Provider,
+            result.Descriptions.Select(x => new LocalizedDescription(x.languageCode, x.longDescription, x.shortDescription)),
+            result.Documents.GroupBy(d => d.documentTypeId).ToDictionary(g => g.Key, g => g.Select(d => new DocumentData(d.documentId, d.documentName))),
+            result.ProviderUri ?? Constants.ErrorString,
+            result.ContactEmail,
+            result.ContactNumber
+
```
Comment (Phil91) [inline src/marketplace/Services.Service/BusinessLogic/ServiceReleaseBusinessLogic.cs:80]: Please remove empty line

File: src/portalbackend/PortalBackend.DBAccess/Repositories/OfferRepository.cs [hunk 3]
```diff
@@ -636,4 +636,22 @@ sorting switch
...
             .SingleOrDefaultAsync();
+
+    ///<inheritdoc/>
+    public Task<ServiceDetailsData?> GetServiceDetailsByIdAsync(Guid serviceId) =>
+        _context.Offers.AsNoTracking()
+            .Where(service => service.Id == serviceId && service.OfferTypeId == OfferTypeId.SERVICE)
+            .Select(offer => new ServiceDetailsData(
+                offer.Id,
+                offer.Name,
+                offer.ServiceDetails.Select(x => x.ServiceTypeId),
```
Comment (Phil91) [inline src/portalbackend/PortalBackend.DBAccess/Repositories/OfferRepository.cs:647]: In the task it looks like we should return the label of ServiceType here

File: src/portalbackend/PortalBackend.DBAccess/Repositories/StaticDataRepository.cs [hunk 4]
```diff
@@ -83,4 +83,10 @@ public StaticDataRepository(PortalDbContext portalDbContext)
                     .Where(identifier => identifier.BpdmIdentifierId != null && bpdmIdentifierIds.Contains(identifier.BpdmIdentifierId.Value))
                     .Select(identifier => new ValueTuple<BpdmIdentifierId,UniqueIdentifierId>(identifier.BpdmIdentifierId!.Value, identifier.UniqueIdentifierId))))
             .SingleOrDefaultAsync();
+    
+    ///<inheritdoc />
+    public IAsyncEnumerable<ServiceTypeData> GetServiceTypeData()=>
+        _dbContext.ServiceTypes.AsNoTracking()
+            .Select(serviceType => new ServiceTypeData((int)serviceType.Id, serviceType.Label))
```
Comment (Phil91) [inline src/portalbackend/PortalBackend.DBAccess/Repositories/StaticDataRepository.cs:90]: the first parameter of ServiceTypeData should be the enum value, in my understanding of type ServiceTypeId

File: tests/marketplace/Services.Service.Tests/BusinessLogic/ServiceReleaseBusinessLogicTest.cs [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+        result.Should().BeOfType<ServiceData>();
+        result.Title.Should().Be("ServiceTest");
+        result.Provider.Should().Be("TestProvider");
+        result.ProviderUri.Should().Be("TestProviderUri");
+        result.ContactEmail.Should().Be("test@gmail.com");
+        result.ContactNumber.Should().Be("6754321786");
+    }
+
+    [Fact]
+    public async Task GetServiceDetailsByIdAsync_WithInvalidOfferStatus_ThrowsException()
```
Comment (Phil91) [inline tests/marketplace/Services.Service.Tests/BusinessLogic/ServiceReleaseBusinessLogicTest.cs:118]: Please add a test for the no service existing case as well

**Review by Phil91 — CHANGES_REQUESTED** [review]

File: src/portalbackend/PortalBackend.DBAccess/Repositories/OfferRepository.cs (hunk 3 above)
Comment (qxz2mqe) [reply src/portalbackend/PortalBackend.DBAccess/Repositories/OfferRepository.cs]: fixed

File: tests/marketplace/Services.Service.Tests/BusinessLogic/ServiceReleaseBusinessLogicTest.cs (hunk 5 above)
Comment (qxz2mqe) [reply tests/marketplace/Services.Service.Tests/BusinessLogic/ServiceReleaseBusinessLogicTest.cs:118]: fixed

**Review by Phil91 — APPROVED** [review]

**Pull request merged**